### PR TITLE
Redo build system in cmake

### PR DESCRIPTION
**Pull request by jagerman**

### CODE REVIEW

**Comment by despair86** [comment]
why

**Comment by notlesh** [comment]
> why

Because `qmake` is moderately shitty, and also because we're using `cmake` for `lokinet`

**Review by neuroscr — COMMENTED** [review]
You missed the svg lib for mac

File: README.md [hunk 1]
```diff
@@ -19,17 +19,29 @@ Then build and run:
...
+
+The above commands work to build a working executable, however if you also want to bundle it into a
+`.app` bundle invoke cmake as:
+
+    cmake -DMACOS_APP=ON ..
+
+which will produce a lokinet-gui.app executable app bundle directory containing the binary,
+resources, and dependencies.
+
+Note that building this way additionally requires imagemagick installed and built with SVG support.
```
Comment (notlesh) [inline README.md:43]: Apple also has a built in utility for creating a ICNS file/dir/whatever from a PNG, so we could avoid this dep by maintaining a PNG version of the icon. That's probably not worth it unless the `imagemagick` dep is problematic.

File: tools/mk-icns.sh [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+for size in 16 32 64 128 256 512 1024; do
+    convert -background none -resize "${size}x${size}" "$svg" -strip "png32:${outdir}/icon_${size}x${size}.png"
+done
+mv "${outdir}/icon_1024x1024.png" "${outdir}/icon_512x512@2x.png"
+for size in 16 32 128 256; do
+    double=$((size * 2))
+    cp "${outdir}/icon_${double}x${double}.png" "${outdir}/icon_${size}x${size}@2x.png"
+done
+
+iconutil -c icns "${outdir}"
```
Comment (notlesh) [inline tools/mk-icns.sh:20]: Is there anything to clean up here?

File: CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+if(APPLE)
+    option(MACOS_APP "Build a LokinetGUI.app bundle instead of a binary" OFF)
+    set(MACOS_SIGN ""
+        CACHE STRING "enable codesigning -- use a 'Apple Distribution' key (or key description) from `security find-identity -v`")
+
+    target_sources(lokinet-gui PRIVATE src/process/MacOSLokinetProcessManager.cpp)
+
+    if(MACOS_APP)
+        add_custom_command(OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/lokinet-gui.icns
+            COMMAND ${CMAKE_CURRENT_SOURCE_DIR}/tools/mk-icns.sh ${CMAKE_CURRENT_SOURCE_DIR}/res/images/icon.svg ${CMAKE_CURRENT_BINARY_DIR}/lokinet-gui.icns
```
Comment (notlesh) [inline CMakeLists.txt:51]: Will this be invoked every time, or only when the `icns` file/dir/whatever needs to be generated?

File: README.md (hunk 1 above)
Comment (jagerman) [reply README.md:43]: I think it's fine; imagemagick is pretty easily available.

File: tools/mk-icns.sh (hunk 2 above)
Comment (jagerman) [reply tools/mk-icns.sh:20]: I mean, it's a build directory...

File: CMakeLists.txt (hunk 3 above)
Comment (jagerman) [reply CMakeLists.txt:51]: The latter.

File: tools/mk-icns.sh (hunk 2 above)
Comment (notlesh) [reply tools/mk-icns.sh:20]: Fair enough

**Pull request merged**